rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -348,6 +348,35 @@ object ProcNormalizeMatcher {
         ProcVisitOutputs(input.par.prepend(foldedNew), bodyResult.knownFree)
       }
 
+      case p: PMatch => {
+        import scala.collection.JavaConverters._
+
+        val targetResult = normalizeMatch(p.proc_, input)
+        val cases = p.listcase_.asScala.toList.map {
+          case ci: CaseImpl => (ci.proc_1, ci.proc_2)
+          case _            => throw new Error("Unexpected Case implementation.")
+        }
+
+        val initAcc = (List[Tuple2[Par, Par]](), input.knownFree)
+        val casesResult = (initAcc /: cases) { (acc, caseImpl) =>
+          caseImpl match {
+            case (pattern, caseBody) => {
+              val patternResult =
+                normalizeMatch(pattern,
+                               ProcVisitInputs(Par(), input.env, DebruijnLevelMap[VarSort]()))
```

Comment:
Patterns also cannot reference the environment.

Reply:
Done